### DISCUSSION TITLE
ninja: build stopped: subcommand failed. macOS下编译出现的问题有人遇见过吗

### DISCUSSION THREAD

**Discussion by youngqj** [apache/doris#21101] [RFC]
[0/2] Re-checking globbed directories...
[47/1126] Building CXX object CMakeFiles/pch.dir/cmake_pch.hxx.pch
FAILED: CMakeFiles/pch.dir/cmake_pch.hxx.pch
ccache /usr/local/opt/llvm@16/bin/clang++ -DBOOST_STACKTRACE_USE_NOOP -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_LIBCPP -DUSE_LIBHDFS3 -I/Users/yangqijun/Downloads/doris-master/be/src/apache-orc/c++/include -I/Users/yangqijun/Downloads/doris-master/be/build_Release/src/apache-orc/c++/include -I/Users/yangqijun/Downloads/doris-master/be/build_Release/src/clucene/src/shared -I/Users/yangqijun/Downloads/doris-master/be/src/clucene/src/core -I/Users/yangqijun/Downloads/doris-master/be/src/clucene/src/shared -I/Users/yangqijun/Downloads/doris-master/be/src/clucene/src/contribs-lib -I/Users/yangqijun/Downloads/doris-master/be/src -I/Users/yangqijun/Downloads/doris-master/be/test -I/Library/Java/JavaVirtualMachines/openjdk-8.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/openjdk-8.jdk/Contents/Home/include/darwin -isystem /Users/yangqijun/Downloads/doris-master/be/../gensrc/build -isystem /Users/yangqijun/Downloads/doris-master/thirdparty/installed/include -isystem /Users/yangqijun/Downloads/doris-master/thirdparty/installed/gperftools/include -O3 -DNDEBUG  -O3 -DNDEBUG -std=gnu++20 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk -mmacosx-version-min=12.6   -D OS_MACOSX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fcolor-diagnostics -Wpedantic -Wunused -Wunused-command-line-argument -Wunused-exception-parameter -Wunused-volatile-lvalue -Wunused-template -Wunused-member-function -Wunused-macros -Wno-vla-extension -stdlib=libc++ -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -DUSE_JEMALLOC -DENABLE_STACKTRACE -Xclang -fno-pch-timestamp -Winvalid-pch -fpch-instantiate-templates -Xclang -emit-pch -Xclang -include -Xclang /Users/yangqijun/Downloads/doris-master/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx -x c++-header -MD -MT CMakeFiles/pch.dir/cmake_pch.hxx.pch -MF CMakeFiles/pch.dir/cmake_pch.hxx.pch.d -o CMakeFiles/pch.dir/cmake_pch.hxx.pch -c /Users/yangqijun/Downloads/doris-master/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx.cxx
In file included from <built-in>:1:
In file included from /Users/yangqijun/Downloads/doris-master/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx:5:
/Users/yangqijun/Downloads/doris-master/be/src/pch/pch.h:499:10: error: expected "FILENAME" or <FILENAME>
#include j<jni.h>
         ^
1 error generated.
[49/1126] Generating bitunpack_sse.o
ninja: build stopped: subcommand failed.

**Comment by youngqj** [comment]
yangqijun@yangqijundeMacBook-Pro doris-master % USE_AVX2=0  sh build.sh --clean --be --fe
Python 3.11.4
Apache Maven 3.9.2 (c9616018c7a021c1c39be70fb2843d6f5f9b8a1c)
Maven home: /usr/local/Cellar/maven/3.9.2/libexec
Java version: 1.8.0_372, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk@8/1.8.0+372/libexec/openjdk.jdk/Contents/Home/jre
Default locale: zh_CN, platform encoding: US-ASCII
OS name: "mac os x", version: "12.6.5", arch: "x86_64", family: "mac"
cmake version 3.26.4

CMake suite maintained and supported by Kitware (kitware.com/cmake).
ninja 1.11.1
ccache version 4.8.2

**Reply by yagagagaga** [reply]
1. make sure you have update the to latest.
2. download the [thirdparty](https://github.com/apache/doris-thirdparty/releases), decompress it and take it to the `${DORIS_HOME}/thirdparty`
3. run `sh build.sh --clean`
4. run `sh build.sh --be --fe`
5. take a patience

**Reply by youngqj** [reply]
thanks!  
download  from https://github.com/apache/doris-thirdparty/releases/download/automation/doris-thirdparty-source.tgz
and  like this :
cd thirdparty
rm -rf installed

# Intel chips
curl -L https://github.com/apache/doris-thirdparty/releases/download/automation/doris-thirdparty-prebuilt-darwin-x86_64.tar.xz \
    -o - | tar -Jxf -


 and then  run sh build.sh --be  
ninja: build stopped: subcommand failed.

**Reply by youngqj** [reply]
document URL is https://doris.apache.org/docs/dev/install/source-install/compilation-mac/
